azure/converters/identity_test.go

```diff
@@ -38,14 +38,14 @@ var sampleSubjectFactory = []infrav1.UserAssignedIdentity{
 }
 
 var expectedVMSDKObject = map[string]*compute.VirtualMachineIdentityUserAssignedIdentitiesValue{
-	"foo":             {},
-	"bar":             {},
+	"/foo":            {},
```

Comment:
@nader-ziada need your help figuring out if this is indeed what we want... The comment on `UserAssignedIdentitiesToVMSDK` say: 

`// The user identity dictionary key references will be ARM resource ids in the form:
// '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}'.`

So I think we expect only the leading `azure://` to be removed, the rest (including the leading `/`) should be kept there?

Reply:
yes, I believe you are correct to assume we should keep one `/`